Add Debian 8 and Ubuntu 16.04 vagrant boxes

AkshitaJha opened the pull request:
Signed-off-by: Akshita Jha <zenith158@gmail.com>

AkshitaJha committed:
Add Debian 8 and Ubuntu 16.04 vagrant boxes

Signed-off-by: Akshita Jha <zenith158@gmail.com>

dfarrell07 commented on Vagrantfile:
```diff
@@ -478,4 +478,41 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  config.vm.define "debian8_ansible" do |debian8_ansible|
+    # Build Vagrant box based on Debian 8
+    debian8_ansible.vm.box = "debian/jessie64"
+    debian8_ansible.vm.box_version = "=8.5.2"
```
Maybe we should build it based on latest, so this doesn't need to be maintained to stay recent?

AkshitaJha replied:
Done.

dfarrell07 commented on Vagrantfile:
```diff
@@ -478,4 +478,41 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+    # Install ODL using the Ansible provisioner
+    debian8_ansible.vm.provision "ansible" do |ansible|
+      # Path to Ansible playbook that installs ODL using ODL's Ansible role
+      ansible.playbook = "provisioning/all_defaults_playbook.yml"
```
After you address https://github.com/dfarrell07/ansible-opendaylight/pull/28#pullrequestreview-11056029, you'll need to change this to define/use a playbook that sets deb_repo as install method.

AkshitaJha replied:
I have added `provisioning/deb_path_install_playbook.yml` and `provisioning/deb_repo_install_playbook.yml`.

dfarrell07 commented on Vagrantfile:
```diff
@@ -478,4 +478,41 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+    # Install ODL using the Ansible provisioner
+    ubuntu16_ansible.vm.provision "ansible" do |ansible|
+      # Path to Ansible playbook that installs ODL using ODL's Ansible role
+      ansible.playbook = "provisioning/all_defaults_playbook.yml"
```
Same deb_repo as install method feedback as above

dfarrell07 commented on Vagrantfile:
```diff
@@ -478,4 +478,41 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
+  config.vm.define "ubuntu16_ansible" do |ubuntu16_ansible|
+    # Build Vagrant box based on Debian 8
+    ubuntu16_ansible.vm.box = "ubuntu/xenial64"
+    ubuntu16_ansible.vm.box_version = "=20161130.0.0"
```
Same "maybe use latest?" feedback as above

dfarrell07 reviewed (changes requested):
Looks great overall, thanks Akshita! Minor feedback inline.

AkshitaJha committed:
Add Debian 8 and Ubuntu 16.04 vagrant boxes

This commit adds Debian 8 and Ubuntu 16.04 vagrant boxes that install
ODL via it's Ansible role, either via debian repo or .deb local/remote
path.

`provisioning/deb_repo_install_playbook.yml`: installs ODL via debian repository
`provisioning/deb_path_install_playbook.yml`: installs ODL via a .deb local/remote path

Signed-off-by: Akshita Jha <zenith158@gmail.com>

dfarrell07 commented:
Perfect Akshita, thanks! Statically reviewed and `vagrant up`'d/poked around, looks good.
